recipes/protobuf/all/conanfile.py

```diff
@@ -290,6 +294,8 @@ def package_info(self):
         self.cpp_info.filenames["cmake_find_package"] = "Protobuf"
         self.cpp_info.filenames["cmake_find_package_multi"] = "protobuf"
         self.cpp_info.names["pkg_config"] ="protobuf_full_package"
-        self.cpp_info.components["libprotobuf"].build_modules = build_modules
+        for generator in ["cmake_find_package", "cmake_find_package_multi"]:
+            self.cpp_info.components["libprotobuf"].build_modules[generator] = build_modules
         if self.options.lite:
-            self.cpp_info.components["libprotobuf-lite"].build_modules = build_modules
+            for generator in ["cmake_find_package", "cmake_find_package_multi"]:
+                self.cpp_info.components["libprotobuf-lite"].build_modules[generator] = build_modules
```

Comment:
Nop, it's a regression.

Reply:
In any case, build_modules should only be assigned to the _root_ component (check [1.51 changelog](https://docs.conan.io/en/latest/changelog.html?highlight=build_modules#jul-2022) - [this PR](https://github.com/conan-io/conan/pull/11673))

Reply:
`cmake_build_modules` property is properly assigned to the root here.
`build_modules` is the legacy build modules, and it can't be assigned to root when there are components.

Reply:
I can confirm this is the reason why `CONAN_BUILD_MODULES_PATHS_PROTOBUF` is empty for `cmake_paths` (and others `cmake_*` generators too).
I don't really know if those cmake files were supposed to be there in the first place, but the problem is that `protobuf_generate_cpp` (which gets defined there) is pretty essential, so why would they not be?

@pgeler any reasoning for this change?